firebase-common/src/test/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatControllerTest.java

```diff
@@ -0,0 +1,160 @@
+        .getHeartBeatsHeader()
+        .addOnCompleteListener(executor, getOnCompleteListener);
+    String expected =
+        Base64.getEncoder()
+            .encodeToString(
+                "[{\"date\":[\"2015-02-03\"],\"agent\":\"test-agent\",\"version\":\"1\"}]"
+                    .getBytes());
+    assertThat(getOnCompleteListener.await()).isEqualTo(expected);
+  }
+
+  @Test
+  public void generateHeartBeat_twoHeartBeatsSameUserAgent()
+      throws ExecutionException, InterruptedException, JSONException {
+    ArrayList<HeartBeatResult> returnResults = new ArrayList<>();
+    ArrayList<String> dateList = new ArrayList<>();
+    dateList.add("2015-03-02");
+    dateList.add("2015-03-01");
+    returnResults.add(HeartBeatResult.create("test-agent", dateList));
+    when(storage.getAllHeartBeats()).thenReturn(returnResults);
+    heartBeatController
+        .registerHeartBeat()
+        .addOnCompleteListener(executor, storeOnCompleteListener);
+    storeOnCompleteListener.await();
+    heartBeatController
+        .registerHeartBeat()
+        .addOnCompleteListener(executor, storeOnCompleteListener);
+    storeOnCompleteListener.await();
+    verify(storage, times(2)).storeHeartBeat(anyLong(), anyString());
+    heartBeatController
+        .getHeartBeatsHeader()
+        .addOnCompleteListener(executor, getOnCompleteListener);
+    String expected =
+        Base64.getEncoder()
+            .encodeToString(
+                "[{\"date\":[\"2015-03-02\",\"2015-03-01\"],\"agent\":\"test-agent\",\"version\":\"1\"}]"
+                    .getBytes());
+    assertThat(getOnCompleteListener.await()).isEqualTo(expected);
+  }
+
+  @Test
+  public void generateHeartBeat_twoHeartBeatstwoUserAgents()
+      throws ExecutionException, InterruptedException, JSONException {
+    ArrayList<HeartBeatResult> returnResults = new ArrayList<>();
+    returnResults.add(
+        HeartBeatResult.create(
+            "test-agent", new ArrayList<String>(Collections.singleton("2015-03-02"))));
+    returnResults.add(
+        HeartBeatResult.create(
+            "test-agent-1", new ArrayList<String>(Collections.singleton("2015-03-03"))));
+    when(storage.getAllHeartBeats()).thenReturn(returnResults);
+    heartBeatController
+        .registerHeartBeat()
+        .addOnCompleteListener(executor, storeOnCompleteListener);
+    storeOnCompleteListener.await();
+    heartBeatController
+        .registerHeartBeat()
+        .addOnCompleteListener(executor, storeOnCompleteListener);
+    storeOnCompleteListener.await();
+    verify(storage, times(2)).storeHeartBeat(anyLong(), anyString());
+    heartBeatController
+        .getHeartBeatsHeader()
+        .addOnCompleteListener(executor, getOnCompleteListener);
+    String expected =
+        Base64.getEncoder()
+            .encodeToString(
+                "[{\"date\":[\"2015-03-02\"],\"agent\":\"test-agent\",\"version\":\"1\"},{\"date\":[\"2015-03-03\"],\"agent\":\"test-agent-1\",\"version\":\"1\"}]"
```

Comment:
Why is the version part of every object? I think instead we wanted it to be:

```json
{
  "version": "1",
  "heartbeats": []
}
```

Reply:
+1